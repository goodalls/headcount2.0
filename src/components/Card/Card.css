.card {
  border: none;
  border-radius: 25px;
  padding: 20px 10px;
  background-color: white;
  box-shadow: 2px 2px 4px #868686;
  min-height: 352px; }
  .card:hover {
    cursor: pointer;
    background-color: #F5F5F6; }
  .card h3 {
    font-size: 1.2em;
    color: #5BB3C3;
    font-weight: 400;
    text-align: center;
    height: 40px; }
  .card li {
    list-style: none;
    margin: 8px 0; }
  .card .card-data {
    display: flex;
    flex-direction: column;
    align-items: center; }
  .card .compare {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    text-align: center;
    padding: 50px 10px;
    height: 100%;
    width: 100%; }
    .card .compare h3 {
      height: auto;
      margin-bottom: 5px; }
    .card .compare h4 {
      font-weight: 400; }
    .card .compare .districts-avg {
      display: flex;
      align-items: center;
      justify-content: space-between;
      width: 100%;
      padding: 0 10px; }
    .card .compare .arrow {
      height: 34px; }

ul {
  margin-top: 10px; }

li {
  list-style: none;
  margin: 2px 0; }

.low {
  color: #ED6033; }

.high {
  color: #273B7A; }

.selected {
  background-color: #DDDDDE; }

.compare-header {
  margin-bottom: 10px; }

@media screen and (max-width: 850px) {
  .card .compare {
    height: 290px; }
    .card .compare .arrow {
      display: none; }
    .card .compare .districts-avg {
      justify-content: center; }
  .compare-header {
    margin-bottom: 0; } }

@media screen and (max-width: 958px) {
  .card h3 {
    height: auto; } }
