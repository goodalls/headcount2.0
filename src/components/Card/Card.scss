@import "../../variables.scss";

.card {
  border: none;
  border-radius: 25px;
  text-align: center;
  padding: 20px 0px;
  background-color: $card-bg;
  box-shadow: 1px 2px 4px #868686;
}

h3 {
  font-size: 1.2em;
  color: $card-title-color;
}

ul {
  margin-top: 10px;
}

li {
  list-style: none;
}
.low{
  color: $card-low-color;
}

.high {
  color: $card-high-color;
}

.selected {
  background-color: $card-bg-selected;
}