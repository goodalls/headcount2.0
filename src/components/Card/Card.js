import PropTypes from 'prop-types';
import React from 'react';
import './Card.css';
import leftArrow from './back-arrow-circular-symbol.svg';
import rightArrow from './right-arrow-circular-button.svg';

const Card = props => {
  let renderedCard;
  const { isSelected, onSelect } = props;
  let klass = isSelected ? 'card selected' : 'card';

  if (!props.compared) {
    const { location, districtData, id } = props;
    const years = Object.keys(districtData).map((year, index) => {
      const style = districtData[year] >= 0.5 ? 'high' : 'low';
      return (
        <li className={style} key={`year-${index}`}>
          {year}: {districtData[year]}
        </li>
      );
    });

    renderedCard = (
      <div className="card-data">
        <h3>{location}</h3>
        <ul>{years}</ul>
      </div>
    );
  } else {
    const districtNames = Object.keys(props);
    klass = 'card middle';

    renderedCard = (
      <div className="compare">
        <div>
          <h3>{districtNames[1]}</h3>
          <h4>{props[districtNames[1]]}</h4>
        </div>
        <div>
          <h3 className="compare-header">District Comparison</h3>
          <div className="districts-avg">
            <img src={leftArrow} alt="left arrow" className="arrow" />
            <h4>{props.compared}</h4>
            <img src={rightArrow} alt="right arrow" className="arrow" />
          </div>
        </div>
        <div>
          <h3>{districtNames[2]}</h3>
          <h4>{props[districtNames[2]]}</h4>
        </div>
      </div>
    );
  }

  return (
    <div
      id={props.id}
      className={klass}
      onClick={() => {
        onSelect(props.location, props.id, props.isSelected);
      }}
    >
      {renderedCard}
    </div>
  );
};

Card.propTypes = {
  location: PropTypes.string,
  districtData: PropTypes.object,
  onSelect: PropTypes.func,
  id: PropTypes.string,
  isSelected: PropTypes.bool,
  compared: PropTypes.oneOfType([PropTypes.bool, PropTypes.number])
};

export default Card;
