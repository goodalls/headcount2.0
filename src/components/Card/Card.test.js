import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  it('should match the snapshot', () => {
    const mockData = { location: 'Colorado', districtData: { '2004': 0.302 } };
    const renderedComponent = shallow(<Card {...mockData} />);

    expect(renderedComponent).toMatchSnapshot();
  });

  it('should have a class of low if data is below 0.5', () => {
    const mockData = { location: 'Colorado', districtData: { '2004': 0.302 } };
    const renderedComponent = shallow(<Card {...mockData} />);

    expect(renderedComponent.find('li').hasClass('low')).toEqual(true);
  });

  it('should have a class of high if data is above 0.5', () => {
    const mockData = { location: 'Colorado', districtData: { '2004': 0.605 } };
    const renderedComponent = shallow(<Card {...mockData} />);

    expect(renderedComponent.find('li').hasClass('high')).toEqual(true);
  });

  // 1. test that checks for location and years (pass in mockData and check for it)

  it('should have location and years', () => {
    const mockData = { location: 'Colorado', districtData: { '2004': 0.605 } };
    const renderedComponent = shallow(<Card {...mockData} />);

    expect(renderedComponent.find('h3').length).toEqual(1);
    expect(renderedComponent.find('ul').length).toEqual(1);
  });
  // 2. test that checks for the conditional rendering that we are going to do with the compare cards

  it('should have conditional compare rendering if compared is true', () => {
    const mockData = {
      location: 'Colorado',
      districtData: { '2004': 0.605 },
      compared: true
    };
    const renderedComponent = shallow(<Card {...mockData} />);

    expect(renderedComponent.find('h3').length).toEqual(2);
    expect(renderedComponent.find('h4').length).toEqual(1);
    expect(renderedComponent.find('ul').length).toEqual(0);
  });

  // 3. test that checks to see if the class changes when it is selected or whatever we are going to do

  it('should have class of compare when true', () => {
    const mockData = {
      location: 'Colorado',
      districtData: { '2004': 0.605 },
      compared: true
    };
    const renderedComponent = shallow(<Card {...mockData} />);

    expect(renderedComponent.find('div.card').hasClass('compare')).toEqual(true);
  });
});
