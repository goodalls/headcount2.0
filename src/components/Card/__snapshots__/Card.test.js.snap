// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    districtData={
        Object {
            "2004": 0.302,
          }
    }
    location="Colorado"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card"
        id={undefined}
        onClick={[Function]}
>
        <h3>
                Colorado
        </h3>
        <ul>
                <li
                        className="low"
                >
                         
                        2004
                        : 
                        0.302
                </li>
        </ul>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Colorado
</h3>,
          <ul>
            <li
                        className="low"
            >
                         
                        2004
                        : 
                        0.302
            </li>
</ul>,
        ],
        "className": "card",
        "id": undefined,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Colorado",
          },
          "ref": null,
          "rendered": "Colorado",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="low"
>
                 
                2004
                : 
                0.302
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "year-0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "2004",
                  ": ",
                  0.302,
                ],
                "className": "low",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "2004",
                ": ",
                0.302,
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card"
          id={undefined}
          onClick={[Function]}
>
          <h3>
                    Colorado
          </h3>
          <ul>
                    <li
                              className="low"
                    >
                               
                              2004
                              : 
                              0.302
                    </li>
          </ul>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Colorado
</h3>,
            <ul>
              <li
                            className="low"
              >
                             
                            2004
                            : 
                            0.302
              </li>
</ul>,
          ],
          "className": "card",
          "id": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Colorado",
            },
            "ref": null,
            "rendered": "Colorado",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="low"
>
                   
                  2004
                  : 
                  0.302
</li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "year-0",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "2004",
                    ": ",
                    0.302,
                  ],
                  "className": "low",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "2004",
                  ": ",
                  0.302,
                ],
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
